recipes/master.rb

```diff
@@ -8,8 +8,28 @@
 #
 #
 
+if node['salt']['master']['api']['enable']
+  if Chef::Resource::ChefGem.method_defined?(:compile_time)
+    chef_gem 'salt-api' do
+      compile_time true
+    end
+  else
+    chef_gem 'salt-api' do
+      action :nothing
+    end.run_action(:install)
+  end
+end
+
 include_recipe 'salt::_setup'
 
+node.default['salt']['master']['config']['external_auth'] = {
+  'pam' => {
+    node['salt']['master']['api']['user']['name'] => [
```

Comment:
need to add `.to_s` at the end of the node attribute, otherwise it stores a `Chef::Node::ImmutableArray` instead of a `String`